### ISSUE TITLE
Token Gate Server with Padplus Puppet: Error [AssertionError]: Assertion failed

### ISSUE THREAD

**Issue by Sunwc0** [wechaty/wechaty#1992]
客户端发送 `#ding` 消息，服务器会出现下列异常消息。
```
10:53:06 VERB Puppet messagePayload(8698359585062855254)                                                                 [17/1897]
Error [AssertionError]: Assertion failed
    at new goog.asserts.AssertionError (/wechaty/node_modules/google-protobuf/google-protobuf.js:81:876)
    at Object.goog.asserts.doAssertFailure_ (/wechaty/node_modules/google-protobuf/google-protobuf.js:82:257)
    at Object.goog.asserts.assert [as assert] (/wechaty/node_modules/google-protobuf/google-protobuf.js:83:83)
    at jspb.BinaryEncoder.writeUnsignedVarint64 (/wechaty/node_modules/google-protobuf/google-protobuf.js:458:77)
    at jspb.BinaryWriter.writeUnsignedVarint64_ (/wechaty/node_modules/google-protobuf/google-protobuf.js:480:507)
    at jspb.BinaryWriter.writeUint64 (/wechaty/node_modules/google-protobuf/google-protobuf.js:485:471)
    at Function.proto.wechaty.puppet.MessagePayloadResponse.serializeBinaryToWriter (/wechaty/node_modules/@chatie/grpc/dist/gener
ated/wechaty/puppet/message_pb.js:849:12)
    at proto.wechaty.puppet.MessagePayloadResponse.serializeBinary (/wechaty/node_modules/@chatie/grpc/dist/generated/wechaty/pupp
et/message_pb.js:812:47)
    at serialize_wechaty_puppet_MessagePayloadResponse (/wechaty/node_modules/@chatie/grpc/dist/generated/wechaty/puppet_grpc_pb.j
s:460:26)
    at sendUnaryResponse (/wechaty/node_modules/grpc/src/server.js:98:15)
    at sendUnaryData (/wechaty/node_modules/grpc/src/server.js:600:9)
    at Object.<anonymous> (/wechaty/node_modules/wechaty-puppet-hostie/src/server/puppet-implementation.ts:529:16)
    at Generator.next (<anonymous>)
    at fulfilled (/wechaty/node_modules/wechaty-puppet-hostie/dist/src/server/puppet-implementation.js:5:58)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  reportErrorToServer: true,
  messagePattern: 'Assertion failed',
  code: 13
}
```

_Originally posted by @jcai in https://github.com/wechaty/wechaty/issues/1985#issuecomment-641212048_

I have the same error

**Comment by huan** [comment]
I have two questions for this issue:

1. What puppet protocol are you using?
1. Can you reproduce this problem on TS Wechaty?

**Comment by Sunwc0** [comment]
1.WECHATY_PUPPET_PADPLUS

2.14:52:20 VERB IoClient onMessage(Message#Text[🗣Contact<Dacong>]        ding)
14:52:20 VERB PuppetServiceImpl messagePayload()
14:52:20 VERB Puppet messagePayload(344428638119965011)
Error [AssertionError]: Assertion failed
    at new goog.asserts.AssertionError (/wechaty/node_modules/google-protobuf/google-protobuf.js:81:876)
    at Object.goog.asserts.doAssertFailure_ (/wechaty/node_modules/google-protobuf/google-protobuf.js:82:257)
    at Object.goog.asserts.assert [as assert] (/wechaty/node_modules/google-protobuf/google-protobuf.js:83:83)
    at jspb.BinaryEncoder.writeUnsignedVarint64 (/wechaty/node_modules/google-protobuf/google-protobuf.js:458:77)
    at jspb.BinaryWriter.writeUnsignedVarint64_ (/wechaty/node_modules/google-protobuf/google-protobuf.js:480:507)
    at jspb.BinaryWriter.writeUint64 (/wechaty/node_modules/google-protobuf/google-protobuf.js:485:471)
    at Function.proto.wechaty.puppet.MessagePayloadResponse.serializeBinaryToWriter (/wechaty/node_modules/@chatie/grpc/dist/generated/wechaty/puppet/message_pb.js:849:12)
    at proto.wechaty.puppet.MessagePayloadResponse.serializeBinary (/wechaty/node_modules/@chatie/grpc/dist/generated/wechaty/puppet/message_pb.js:812:47)
    at serialize_wechaty_puppet_MessagePayloadResponse (/wechaty/node_modules/@chatie/grpc/dist/generated/wechaty/puppet_grpc_pb.js:460:26)
    at sendUnaryResponse (/wechaty/node_modules/grpc/src/server.js:98:15)
    at sendUnaryData (/wechaty/node_modules/grpc/src/server.js:600:9)
    at Object.<anonymous> (/wechaty/node_modules/wechaty-puppet-hostie/src/server/puppet-implementation.ts:529:16)
    at Generator.next (<anonymous>)
    at fulfilled (/wechaty/node_modules/wechaty-puppet-hostie/dist/src/server/puppet-implementation.js:5:58)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  reportErrorToServer: true,
  messagePattern: 'Assertion failed',
  code: 13
}

控制台能打印别人发送的消息，但是后续就报错

**Comment by huan** [comment]
It seems that the padplus puppet did not follow the typing in the message payloads in some cases.

Please let us know if you have more information about this issue, thanks!

**Comment by Sunwc0** [comment]
起来docker服务后 能正常登录显示已登录信息 但是只要接收消息就会报错
ding-dong-bot.py

Wechaty - INFO - receive <error> event <(<Status.INTERNAL: 13>, 'Assertion failed', None)>
INFO:Wechaty:receive <error> event <(<Status.INTERNAL: 13>, 'Assertion failed', None)>

docker 日志：
14:52:20 VERB EventStreamManager onStreamingCallEnd(callback)
14:52:20 VERB EventStreamManager start() puppet is logged in, emit a login event for downstream
14:52:20 VERB Puppet selfId()
14:52:20 VERB EventStreamManager grpcEmit(EVENT_TYPE_LOGIN[25], {"contactId":"wxid_3zqf0o3ubv3422"})
14:52:20 VERB PuppetServiceImpl contactPayload()
14:52:20 VERB Message static load(344428638119965011)
14:52:20 VERB Message constructor(344428638119965011) for class Message
14:52:20 VERB Message ready()
14:52:20 VERB Puppet messagePayload(344428638119965011)
14:52:20 VERB EventStreamManager grpcEmit(EVENT_TYPE_MESSAGE[2], {"messageId":"344428638119965011"})
14:52:20 VERB IoClient onMessage(Message#Text[🗣Contact<Dacong>]        ding)
14:52:20 VERB PuppetServiceImpl messagePayload()
14:52:20 VERB Puppet messagePayload(344428638119965011)
Error [AssertionError]: Assertion failed
    at new goog.asserts.AssertionError (/wechaty/node_modules/google-protobuf/google-protobuf.js:81:876)
    at Object.goog.asserts.doAssertFailure_ (/wechaty/node_modules/google-protobuf/google-protobuf.js:82:257)
    at Object.goog.asserts.assert [as assert] (/wechaty/node_modules/google-protobuf/google-protobuf.js:83:83)
    at jspb.BinaryEncoder.writeUnsignedVarint64 (/wechaty/node_modules/google-protobuf/google-protobuf.js:458:77)
    at jspb.BinaryWriter.writeUnsignedVarint64_ (/wechaty/node_modules/google-protobuf/google-protobuf.js:480:507)
    at jspb.BinaryWriter.writeUint64 (/wechaty/node_modules/google-protobuf/google-protobuf.js:485:471)
    at Function.proto.wechaty.puppet.MessagePayloadResponse.serializeBinaryToWriter (/wechaty/node_modules/@chatie/grpc/dist/generated/wechaty/puppet/message_pb.js:849:12)
    at proto.wechaty.puppet.MessagePayloadResponse.serializeBinary (/wechaty/node_modules/@chatie/grpc/dist/generated/wechaty/puppet/message_pb.js:812:47)
    at serialize_wechaty_puppet_MessagePayloadResponse (/wechaty/node_modules/@chatie/grpc/dist/generated/wechaty/puppet_grpc_pb.js:460:26)
    at sendUnaryResponse (/wechaty/node_modules/grpc/src/server.js:98:15)
    at sendUnaryData (/wechaty/node_modules/grpc/src/server.js:600:9)
    at Object.<anonymous> (/wechaty/node_modules/wechaty-puppet-hostie/src/server/puppet-implementation.ts:529:16)
    at Generator.next (<anonymous>)
    at fulfilled (/wechaty/node_modules/wechaty-puppet-hostie/dist/src/server/puppet-implementation.js:5:58)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  reportErrorToServer: true,
  messagePattern: 'Assertion failed',
  code: 13
}
14:52:22 VERB PuppetServiceImpl ding()
14:52:22 VERB EventStreamManager grpcEmit(EVENT_TYPE_DONG[3], {"data":"ding-dong"})
14:52:23 VERB EventStreamManager this.onStreamingCallEnd() this.eventStream.on(cancelled) fired with arguments: {}
14:52:23 VERB EventStreamManager connectPuppetEventToStreamingCall() unconnect() 13 callbacks

**Comment by jcai** [comment]
问题一样一样的

**Comment by Sunwc0** [comment]
当我中断程序时抛出异常：
task: <Task pending coro=<Wechaty.init_puppet_event_bridge.<locals>.error_listener() running at /Users/sunwc/Documents/anaconda3/lib/python3.7/site-packages/wechaty/wechaty.py:370>>
sys:1: RuntimeWarning: coroutine 'Wechaty.init_puppet_event_bridge.<locals>.error_listener' was never awaited

**Comment by Sunwc0** [comment]
上午刚拉的镜像 然后pad的token  ，用python(wechaty版本0.5.dve1)的example ding-dong-bot.py 登录微信，然后用别的微信给登录的微信发消息就会出现这个断言错误 ，但是服务器上能够收到发送的消息 没有其他多余的代码操作 ，debug的话根本进不到onMessage里去

只要给登录的微信发消息就是这个错误 但是收消息都能在verbose日志中打印出来

**Comment by inkpool** [comment]
I got the same problem here.

**Comment by inkpool** [comment]
Any update or ETA for this problem?

**Comment by jcai** [comment]
@Sunwc0  you can try

```
docker run -e "WECHATY_PUPPET_PADPLUS_TOKEN=puppet_padplus_xxxxxxxxxxx" 
    docker.pkg.github.com/wechaty/scala-wechaty/ding-dong-bot:0.0.16-SNAPSHOT
```

**Comment by landgreenland** [comment]
搞定了吗  这个问题

**Comment by huan** [comment]
This bug was confirmed and has been fixed on wechaty@0.41.26 or later versions.
